File: src/main/java/com/chattriggers/ctjs/minecraft/wrappers/objects/threading/WrappedThread.java
package com.chattriggers.ctjs.minecraft.wrappers.objects.threading;

import com.chattriggers.ctjs.utils.console.Console;

public class WrappedThread extends Thread {
    public WrappedThread(Runnable task) {
        super(() -> {
            try {
                task.run();
            } catch (Exception e) {
                Console.getInstance().printStackTrace(e);

            }
        });
    }
}

File: src/main/kotlin/com/chattriggers/ctjs/minecraft/wrappers/objects/threading/WrappedThread.kt
package com.chattriggers.ctjs.minecraft.wrappers.objects.threading

import com.chattriggers.ctjs.engine.langs.js.JSContextFactory
import com.chattriggers.ctjs.engine.langs.js.JSLoader
import com.chattriggers.ctjs.printTraceToConsole
import org.mozilla.javascript.Context

class WrappedThread(task: Runnable) : Thread({
    try {
        JSContextFactory.enterContext()
        task.run()
        Context.exit()
    } catch (e: Throwable) {
        e.printTraceToConsole(JSLoader.console)
    }
})
